broker.go

```diff
@@ -31,17 +29,15 @@ func newBroker(
 type broker struct {
 	client     kinesisiface.KinesisAPI
 	streamName string
-	shardc     chan *kinesis.Shard
 	logger     Logger
-
-	shardMu sync.Mutex
-	shards  map[string]*kinesis.Shard
+	shardMu    sync.Mutex
+	shards     map[string]*kinesis.Shard
 }
 
-// start is a blocking operation which will loop and attempt to find new
+// Start is a blocking operation which will loop and attempt to find new
 // shards on a regular cadence.
-func (b *broker) start(ctx context.Context) {
-	b.findNewShards()
+func (b *broker) Start(ctx context.Context, shardc chan string) {
```

Comment:
we could potentially adjust this func signature to have it return `chan string`

```go
func (b *broker) Start(ctx context.Context) <-chan string {
```

Reply:
Let me investigate that.